Csc invoked in XamlPrecompile doesn't get Nullable passed

<!-- This is a template that helps us provide quicker feedback. Please use any relevant sections and delete anything you don't need. -->

### Issue Description
Csc invoked in XamlPrecompile doesn't get Nullable passed, and results in warning for NRT disabled when the project file has `<Nullable>enable</Nullable>`.

Root cause of https://github.com/dotnet/roslyn/issues/46420 .

### Steps to Reproduce
1. Create a WinUI 3 template application.
2. Ensure there is some xaml.
3. Add `<Nullable>enable</Nullable>` in project file.
4. Write some code with nullable reference type, such as a `string?` field in `App`.

### Expected Behavior
Build without warning.

### Actual Behavior
The IDE reports no warning. Build output reports warning CS8632: nullable reference types can only be used under `#nullable enable`.

### Analysis
https://github.com/dotnet/roslyn/blob/4f1711c578fd260221f968696ab07b125581b9e1/src/Compilers/Core/MSBuildTask/Microsoft.CSharp.Core.targets#L113

The `<Nullable>` property is missing in `Microsoft.CSharp.CurrentVersion.targets`:

https://github.com/dotnet/msbuild/blob/ec6ed832e6a4adf94a2e6ff7e2f34d8e2da1cc05/src/Tasks/Microsoft.CSharp.CurrentVersion.targets#L277-L281

### Versions & Configurations
Visual Studio 16.8 - 16.9 Preview 3

### Attach a binlog
(Not necessary I think)

--- a/src/Tasks/Microsoft.CSharp.CurrentVersion.targets
+++ b/src/Tasks/Microsoft.CSharp.CurrentVersion.targets
@@ -278,6 +278,7 @@ Copyright (C) Microsoft Corporation. All rights reserved.
               NoLogo="$(NoLogo)"
               NoStandardLib="$(NoCompilerStandardLib)"
               NoWin32Manifest="$(NoWin32Manifest)"
+              Nullable="$(Nullable)"
               Optimize="$(Optimize)"
               OutputAssembly="@(XamlIntermediateAssembly)"
               PdbFile="$(PdbFile)" 

